lib/livebook/session/data.ex

```diff
@@ -877,15 +882,19 @@ defmodule Livebook.Session.Data do
   end
 
   def apply_operation(data, {:set_notebook_hub, _client_id, id}) do
-    with {:ok, hub} <- Hubs.fetch_hub(id) do
+    result =
+      case Hubs.fetch_hub(id) do
+        {:ok, _} = result -> result
+        :error -> Hubs.fetch_offline_hub(id)
+      end
```

Comment:
Yeah, i don't like that we are having to apply the same logic in many places. :(